### PR TITLE
MF-118 - Replace MF_NATS_URL with MF_BROKER_URL

### LINKED ISSUE #118: mainflux cannot be deployed on kubernetes v1.23+
I manually modified the following files to adapt to v1.23+
but after helm install, mainflux-mqtt-0 not running, how can i solve it
env: Rancher rke version v1.3.11

charts/jaeger-operator/crds/crd.yaml

```
apiVersion: apiextensions.k8s.io/v1
kind: CustomResourceDefinition
metadata:
  name: jaegers.jaegertracing.io
spec:
  conversion:
    strategy: None
  group: jaegertracing.io
  names:
    kind: Jaeger
    listKind: JaegerList
    plural: jaegers
    singular: jaeger
  scope: Namespaced
  versions:
  - name: v1
    schema:
      openAPIV3Schema:
        description: cluster
        type: object
    served: true
    storage: true
```

modify postgresql and redis version in Chart.yaml

```
- name: postgresql
  version: "10.14.3"
  repository: "@bitnami"
- name: redis
  version: "15.7.0"
  repository: "@bitnami"

```

```
$ kubectl get po -n mf
NAME                                               READY   STATUS             RESTARTS         AGE
mainflux-adapter-coap-7477d8945-k72rf              0/1     CrashLoopBackOff   10 (3m56s ago)   30m
mainflux-adapter-http-677d4b8c9d-4zgft             0/1     CrashLoopBackOff   10 (3m27s ago)   30m
mainflux-auth-6968cb6649-5pnmz                     1/1     Running            3 (29m ago)      30m
mainflux-auth-6968cb6649-k9g4l                     1/1     Running            3 (29m ago)      30m
mainflux-auth-6968cb6649-zpxft                     1/1     Running            3 (29m ago)      30m
mainflux-envoy-c4fcdb564-cmzfz                     1/1     Running            0                30m
mainflux-envoy-c4fcdb564-ln7c8                     1/1     Running            0                30m
mainflux-envoy-c4fcdb564-z7tgm                     1/1     Running            0                30m
mainflux-grafana-5ddbcbb6fb-grb9m                  1/1     Running            0                30m
mainflux-jaeger-operator-74ff955db6-b8zbw          1/1     Running            0                30m
mainflux-jaeger-operator-jaeger-684df956cd-bpq2r   1/1     Running            0                30m
mainflux-keto-84997fd4f-lpmgh                      1/1     Running            0                30m
mainflux-mqtt-0                                    1/2     CrashLoopBackOff   10 (3m32s ago)   30m
mainflux-nats-0                                    1/1     Running            0                30m
mainflux-nats-1                                    1/1     Running            0                29m
mainflux-nats-2                                    1/1     Running            0                29m
mainflux-postgresqlauth-0                          1/1     Running            0                30m
mainflux-postgresqlketo-0                          1/1     Running            0                30m
mainflux-postgresqlthings-0                        1/1     Running            0                30m
mainflux-postgresqlusers-0                         1/1     Running            0                30m
mainflux-redis-auth-master-0                       1/1     Running            0                30m
mainflux-redis-auth-replicas-0                     1/1     Running            0                30m
mainflux-redis-auth-replicas-1                     1/1     Running            0                29m
mainflux-redis-auth-replicas-2                     1/1     Running            0                28m
mainflux-redis-mqtt-master-0                       1/1     Running            0                30m
mainflux-redis-mqtt-replicas-0                     1/1     Running            0                30m
mainflux-redis-mqtt-replicas-1                     1/1     Running            0                29m
mainflux-redis-mqtt-replicas-2                     1/1     Running            0                28m
mainflux-redis-streams-master-0                    1/1     Running            0                30m
mainflux-redis-streams-replicas-0                  1/1     Running            0                30m
mainflux-redis-streams-replicas-1                  1/1     Running            0                29m
mainflux-redis-streams-replicas-2                  1/1     Running            0                28m
mainflux-things-7df86f8fd5-j4j9x                   1/1     Running            4 (29m ago)      30m
mainflux-things-7df86f8fd5-lrt9m                   1/1     Running            3 (29m ago)      30m
mainflux-things-7df86f8fd5-qk85h                   1/1     Running            3 (29m ago)      30m
mainflux-ui-6568c7d95-hzsmd                        1/1     Running            0                30m
mainflux-users-5764fdcfd5-qx5bq                    1/1     Running            3 (29m ago)      30m
```

```
$ kubectl logs -n mf mainflux-mqtt-0 -c mainflux-adapter-mqtt
2022/07/27 07:53:24 The binary was build using Nats as the message broker
{"level":"info","message":"gRPC communication is not encrypted","ts":"2022-07-27T07:53:24.550574065Z"}
{"level":"error","message":"Failed to connect to message broker: nats: no servers available for connection","ts":"2022-07-27T07:53:24.552383451Z"}
```

### PR DESCRIPTION
**Pull request by chen-dq**
* issus: https://github.com/mainflux/devops/issues/118

Signed-off-by: chen.dq <chen.dq@neusoft.com>

Resolves #118

### CODE REVIEW

**Review by manuio — CHANGES_REQUESTED** [review]
LGTM

**Review by manuio — APPROVED** [review]

File: charts/mainflux/templates/adapter_coap-deployment.yaml [hunk 1]
```diff
@@ -20,7 +20,7 @@ spec:
         - env:
             - name: MF_JAEGER_URL
               value: {{ .Release.Name }}-jaeger-operator-jaeger-agent:{{ .Values.defaults.jaegerPort }}
-            - name: MF_NATS_URL
+            - name: MF_BROKER_URL
               value: nats://{{ .Release.Name }}-nats-client:{{ .Values.defaults.natsPort }}
```
Comment (0x6f736f646f) [inline charts/mainflux/templates/adapter_coap-deployment.yaml:24]: change it to a single variable i.e 

```
- name: MF_NATS_URL
  value: nats://{{ .Release.Name }}-nats-client:{{ .Values.defaults.messageBrokerUrl }}
```

File: charts/mainflux/templates/adapter_http-deployment.yaml [hunk 2]
```diff
@@ -24,7 +24,7 @@ spec:
               value: {{ default .Values.defaults.logLevel .Values.adapter_http.logLevel }}
             - name: MF_HTTP_ADAPTER_PORT
               value: "{{ .Values.adapter_http.httpPort }}"
-            - name: MF_NATS_URL
+            - name: MF_BROKER_URL
               value: nats://{{ .Release.Name }}-nats-client:{{ .Values.defaults.natsPort }}
```
Comment (0x6f736f646f) [inline charts/mainflux/templates/adapter_http-deployment.yaml:28]: Same to here

File: charts/mainflux/templates/adapter_lora-deployment.yaml [hunk 3]
```diff
@@ -23,7 +23,7 @@ spec:
               value: "{{ default .Values.adapter_lora.httpPort }}"
             - name: MF_LORA_ADAPTER_LOG_LEVEL
               value: {{ default .Values.defaults.logLevel .Values.adapter_lora.logLevel }}
-            - name: MF_NATS_URL
+            - name: MF_BROKER_URL
               value: nats://{{ .Release.Name }}-nats-client:{{ .Values.defaults.natsPort }}
```
Comment (0x6f736f646f) [inline charts/mainflux/templates/adapter_lora-deployment.yaml:27]: Same to here

File: charts/mainflux/templates/adapter_mqtt-statefulstet.yaml [hunk 4]
```diff
@@ -179,7 +179,7 @@ spec:
               value: "{{ .Values.mqtt.adapter.wsPort }}"
             - name: MF_MQTT_ADAPTER_ES_URL
               value: {{ .Release.Name }}-redis-streams-master:{{ .Values.mqtt.redisESPort }}
-            - name: MF_NATS_URL
+            - name: MF_BROKER_URL
               value: nats://{{ .Release.Name }}-nats-client:{{ .Values.defaults.natsPort }}
```
Comment (0x6f736f646f) [inline charts/mainflux/templates/adapter_mqtt-statefulstet.yaml:183]: Same to here

File: charts/mainflux/templates/adapter_opcua-deployment.yaml [hunk 5]
```diff
@@ -23,7 +23,7 @@ spec:
               value: "{{ .Values.adapter_opcua.httpPort }}"
             - name: MF_OPCUA_ADAPTER_LOG_LEVEL
               value: {{ default .Values.defaults.logLevel .Values.adapter_opcua.logLevel }}
-            - name: MF_NATS_URL
+            - name: MF_BROKER_URL
               value: nats://{{ .Release.Name }}-nats-client:{{ .Values.defaults.natsPort }}
```
Comment (0x6f736f646f) [inline charts/mainflux/templates/adapter_opcua-deployment.yaml:27]: Same to here

File: charts/mainflux/templates/influxdbwriter-deployment.yaml [hunk 6]
```diff
@@ -32,7 +32,7 @@ spec:
         - env:
             - name: MF_INFLUX_WRITER_LOG_LEVEL
               value: debug
-            - name: MF_NATS_URL
+            - name: MF_BROKER_URL
               value: nats://{{ .Release.Name }}-nats-client:{{ .Values.defaults.natsPort }}
```
Comment (0x6f736f646f) [inline charts/mainflux/templates/influxdbwriter-deployment.yaml:36]: Same to here

File: charts/mainflux/templates/mongodbwriter-deployment.yaml [hunk 7]
```diff
@@ -32,7 +32,7 @@ spec:
         - env:
             - name: MF_MONGO_WRITER_LOG_LEVEL
               value: debug
-            - name: MF_NATS_URL
+            - name: MF_BROKER_URL
               value: nats://{{ .Release.Name }}-nats-client:{{ .Values.defaults.natsPort }}
```
Comment (0x6f736f646f) [inline charts/mainflux/templates/mongodbwriter-deployment.yaml:36]: Same to here

File: charts/mainflux/templates/notifier_smtp-deployment.yaml [hunk 8]
```diff
@@ -87,7 +87,7 @@ spec:
               value: {{ default .Values.defaults.logLevel .Values.notifier_smtp.logLevel }}
             - name: MF_AUTH_GRPC_URL
               value: {{ .Release.Name }}-envoy:{{ .Values.authn.grpcPort }} 
-            - name: MF_NATS_URL
+            - name: MF_BROKER_URL
               value: nats://{{ .Release.Name }}-nats-client:{{ .Values.defaults.natsPort }}
```
Comment (0x6f736f646f) [inline charts/mainflux/templates/notifier_smtp-deployment.yaml:91]: Same to here

File: charts/mainflux/templates/twins-deployment.yaml [hunk 9]
```diff
@@ -31,7 +31,7 @@ spec:
               value: "{{.Values.twins.dbPort }}"
             - name: MF_TWINS_CHANNEL_ID
               value: {{.Values.twins.channelId }}
-            - name: MF_NATS_URL
+            - name: MF_BROKER_URL
               value: nats://{{ .Release.Name }}-nats-client:{{ .Values.defaults.natsPort }}
```
Comment (0x6f736f646f) [inline charts/mainflux/templates/twins-deployment.yaml:35]: Same to here

File: charts/mainflux/templates/adapter_coap-deployment.yaml (hunk 1 above)
Comment (manuio) [reply charts/mainflux/templates/adapter_coap-deployment.yaml:24]: You probably mean
```
- name: MF_BROKER_URL
  value: {{ .Values.defaults.messageBrokerUrl }}
 ```

**Review by 0x6f736f646f — CHANGES_REQUESTED** [review]
Also Rename `natsPort: 4222` to `messageBrokerUrl` with the correct value

File: charts/mainflux/templates/adapter_coap-deployment.yaml (hunk 1 above)
Comment (0x6f736f646f) [reply charts/mainflux/templates/adapter_coap-deployment.yaml:24]: Yes

**Comment by chen-dq** [comment]
> natsPort: 4222

**Comment by 0x6f736f646f** [comment]
@manuio Should we make the `messageBrokerUrl` be configurable to nats, rabbitmq and possibly kafka or just use nats as the default broker

**Comment by manuio** [comment]
> @manuio Should we make the `messageBrokerUrl` be configurable to nats, rabbitmq and possibly kafka or just use nats as the default broker

@0x6f736f646f I'd say that we should make it configurable and remove any hardcoded value from the charts. Not sure what to do with the `{{ .Release.Name }}`. Maybe we can have something like `{{Values.defaults.msgBrokerProtocol}}://{{ .Release.Name }}-{{msgBrokerUrl}}:{{ .Values.defaults.msgBrokerPort }}`. But it's a bit confusing and unreadable compared to `{{messageBrokerUrl}}`.

File: charts/mainflux/templates/adapter_coap-deployment.yaml (hunk 1 above)
Comment (arvindh123) [inline charts/mainflux/templates/adapter_coap-deployment.yaml:24]: @chen-dq , 
The nats broker in the values.yaml file is referred to as `nats`, which means that the hostname of the nats broker will be {{ .Release.Name}}-nats. Can you please update this change in all the relevant locations?

```suggestion
            - name: MF_BROKER_URL
              value: nats://{{ .Release.Name }}-nats:{{ .Values.defaults.natsPort }}   
```

**Comment by arvindh123** [comment]
I also agree with @manuio  point about not changing this until we have support for all brokers.

**Review by arvindh123 — APPROVED** [review]
LGTM

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**